crates/bevy_ui/src/layout/mod.rs

```diff
@@ -330,14 +340,18 @@ pub fn ui_layout_system(
                     );
                     continue;
                 };
+                ui_surface.no_camera = false;
                 let layout_info = camera_layout_info
                     .entry(camera_entity)
                     .or_insert_with(|| calculate_camera_layout_info(camera));
                 layout_info.root_nodes.push(entity);
             }
             None => {
                 if cameras.is_empty() {
-                    warn!("No camera found to render UI to. To fix this, add at least one camera to the scene.");
+                    if !ui_surface.no_camera {
+                        ui_surface.no_camera = true;
```

Comment:
IMO the `no_camera` flag is not needed and just adds an extra layer. I tested it with this diff below applied to your branch and it still passed the test. But the else upsert_node below with the default style and layout is a good catch. :+1:

If we are worried about spamming the console we could use a `warn_once!` instead.

```dif
Index: crates/bevy_ui/src/layout/mod.rs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/crates/bevy_ui/src/layout/mod.rs b/crates/bevy_ui/src/layout/mod.rs
--- a/crates/bevy_ui/src/layout/mod.rs	(revision 8bbf91532c1fcdef9949ae021f3c3531c970c865)
+++ b/crates/bevy_ui/src/layout/mod.rs	(date 1711922950048)
@@ -63,7 +63,6 @@
     camera_entity_to_taffy: EntityHashMap<EntityHashMap<taffy::node::Node>>,
     camera_roots: EntityHashMap<Vec<RootNodePair>>,
     taffy: Taffy,
-    no_camera: bool,
 }
 
 fn _assert_send_sync_ui_surface_impl_safe() {
@@ -91,7 +90,6 @@
             camera_entity_to_taffy: Default::default(),
             camera_roots: Default::default(),
             taffy,
-            no_camera: false,
         }
     }
 }
@@ -340,7 +338,6 @@
                     );
                     continue;
                 };
-                ui_surface.no_camera = false;
                 let layout_info = camera_layout_info
                     .entry(camera_entity)
                     .or_insert_with(|| calculate_camera_layout_info(camera));
@@ -348,10 +345,7 @@
             }
             None => {
                 if cameras.is_empty() {
-                    if !ui_surface.no_camera {
-                        ui_surface.no_camera = true;
-                        warn!("No camera found to render UI to. To fix this, add at least one camera to the scene.");
-                    }
+                    warn!("No camera found to render UI to. To fix this, add at least one camera to the scene.");
                 } else {
                     warn!(
                         "Multiple cameras found, causing UI target ambiguity. \
```

Reply:
That makes perfect sense, I'll make the commit with that changes! Thank you for your help!